docs/ProgrammingCryptol/highAssurance/HighAssurance.tex

```diff
@@ -174,6 +174,38 @@ \subsection{Capturing test vectors}
   sure the tweak to the above example (removal of dummy single formal
   parameter) works.}
 
+%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+\subsection{Generating test vectors}
+\label{sec:gentestvec}
+
+Cryptol has the ability to generate test vectors for non-polymorphic functions (there is limited support for polymorphic types) using the \texttt{:dumptests} REPL command.
+When \texttt{:dumptests} is given a file to dump the output and a function in scope, it will randomly generate inputs for the function and run them for you.
+\begin{Verbatim}
+  :dumptests <results file> <function>
+  :dumptests results.txt myFunction
+\end{Verbatim}
+The results file will contain a tab delimited file in which the first column contains the output, and each column after contains the inputs to the function in order.
+You can set the number of tests you want to generate in your test vector by using the \texttt{:set tests=100} command. Additionally, you can also change base using \texttt{:set base=2}.
+This is because the command \texttt{:dumptests} uses the same settings for random generation that is used in \ref{sec:quickcheck}. 
+The example below shows \texttt{:dumptests} run on an instance of a function polymorphic \texttt{f} which has two inputs. 
+\begin{code}
+  f : {n} (fin n, n >= 2) => [n] -> [n] -> [n]
```

Comment:
suggestion: I would really like it if one of these parameters was different, so that the reader can visually see how the type maps onto the result file below. Maybe
```
f : {n} (fin n, n >= 2) => [n] -> [n] -> [2 * n] 
f x y = x # (2 * y)
```